src/pages/Reports/components/QuestionnaireInputs/QuestionnaireInputs.jsx

```diff
@@ -0,0 +1,122 @@
+import React, { useEffect, useState, forwardRef, useImperativeHandle } from 'react';
+import { useTranslation } from 'react-i18next';
+import { toast } from 'react-toastify';
+import UserStore from '../../../../stores/User.store';
+import NodeGroupSelect from '../../../../common/NodeGroupSelect/NodeGroupSelect';
+import UnitSelect from '../../../../common/UnitSelect/UnitSelect';
+import { targetTypes } from '../../../../services/questionnaireSchema.service';
+import NodeService from '../../../../services/node.service';
+import GenericSelect from '../../../../common/GenericSelect/GenericSelect';
+import LabeledInput from '../LabeledInput/LabeledInput';
+import useStyles from '../Input/Input.styles.js';
+
+const QuestionnaireInputs = ({
+  setCanSubmit,
+  unitsOptionalRoles,
+  nodeGroupsRequiredRoles,
+  nodeGroupsOptionalRoles,
+  setDataToSend,
+ }, ref) => {
+  const classes = useStyles();
+  const { t } = useTranslation();
+  const currentUser = UserStore.userProfile;
+  const [selectedNode, setSelectedNode] = useState();
+  const [selectedTarget, setSelectedTarget] = useState();
+  const [selectedNodeGroup, setSelectedNodeGroup] = useState();
+  const [selectedUnit, setSelectedUnit] = useState();
+  const [nodes, setNodes] = useState([]);
+
+  useEffect(() => {
+    setCanSubmit(
+      !(nodeGroupsRequiredRoles.includes(currentUser.role) && !selectedNodeGroup)
```

Comment:
Change to a more intuitive syntax
```suggestion
      !nodeGroupsRequiredRoles.includes(currentUser.role) || selectedNodeGroup
```

Reply:
it is more readable that way

Reply:
How?